tests/integration/thrift_healthcheck_tests.py

```diff
@@ -0,0 +1,190 @@
+    # create baseplate root
+    baseplate = Baseplate()
+    if server_span_observer:
+
+        class TestBaseplateObserver(BaseplateObserver):
+            def on_server_span_created(self, context, server_span):
+                server_span.register(server_span_observer)
+
+        baseplate.register(TestBaseplateObserver())
+
+    # set up the server's processor
+    logger = mock.Mock(spec=logging.Logger)
+    edge_context_factory = make_edge_context_factory()
+    processor = server_spec.Processor(handler)
+    processor = baseplateify_processor(processor, logger, baseplate, edge_context_factory)
+
+    # bind a server socket on an available port
+    server_bind_endpoint = config.Endpoint("127.0.0.1:0")
+    listener = make_listener(server_bind_endpoint)
+    server = make_server(
+        {"max_concurrency": "100", "stop_timeout": "1 millisecond"}, listener, processor
+    )
+
+    # figure out what port the server ended up on
+    server_address = listener.getsockname()
+    server.endpoint = config.Endpoint(f"{server_address[0]}:{server_address[1]}")
+
+    # run the server until our caller is done with it
+    server_greenlet = gevent.spawn(server.serve_forever)
+    try:
+        yield server
+    finally:
+        server_greenlet.kill()
+
+
+@contextlib.contextmanager
+def raw_thrift_client(endpoint):
+    pool = ThriftConnectionPool(endpoint)
+    with pool.connection() as client_protocol:
+        yield BaseplateServiceV2.Client(client_protocol)
+
+
+@contextlib.contextmanager
+def baseplate_thrift_client(endpoint, client_span_observer=None):
+    baseplate = Baseplate(
+        app_config={
+            "baseplate.service_name": "fancy test client",
+            "example_service.endpoint": str(endpoint),
+        }
+    )
+
+    if client_span_observer:
+
+        class TestServerSpanObserver(ServerSpanObserver):
+            def on_child_span_created(self, span):
+                span.register(client_span_observer)
+
+        observer = TestServerSpanObserver()
+
+        class TestBaseplateObserver(BaseplateObserver):
+            def on_server_span_created(self, context, span):
+                span.register(observer)
+
+        baseplate.register(TestBaseplateObserver())
+
+    context = baseplate.make_context_object()
+    trace_info = TraceInfo.from_upstream(
+        trace_id=1234, parent_id=2345, span_id=3456, flags=4567, sampled=True
+    )
+
+    baseplate.configure_context({"example_service": ThriftClient(BaseplateServiceV2.Client)})
+
+    baseplate.make_server_span(context, "example_service.example", trace_info)
+
+    edge_context_factory = make_edge_context_factory()
+    edge_context = edge_context_factory.from_upstream(SERIALIZED_EDGECONTEXT_WITH_VALID_AUTH)
+    edge_context.attach_context(context)
+
+    yield context
+
+
+class GeventPatchedTestCase(unittest.TestCase):
+    def setUp(self):
+        gevent.monkey.patch_socket()
+
+    def tearDown(self):
+        import socket
+
+        reload(socket)
```

Comment:
this feels like a ton of duplication. could these be parameterized in the original module and either imported here or the test suite below just moved there? (i lean towards the latter as this is just a subset of thrift integration tests)